include/i2s.h

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright (c) 2017 Piotr Mienkowski
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public APIs for the I2S (Inter-IC Sound) bus drivers.
+ */
+
+#ifndef __I2S_H__
+#define __I2S_H__
+
+/**
+ * @defgroup i2s_interface I2S Interface
+ * @ingroup io_interfaces
+ * @brief I2S (Inter-IC Sound) Interface
+ *
+ * The I2S API provides support for the standard I2S interface standard as well
+ * as common non-standard extensions such as PCM Short/Long Frame Sync,
+ * Left/Right Justified Data Format.
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * The following #defines are used to configure the I2S controller.
+ */
+
+
+typedef u8_t i2s_fmt_t;
+
+/** Data Format bit field position. */
+#define I2S_FMT_DATA_FORMAT_SHIFT           0
+/** Data Format bit field mask. */
+#define I2S_FMT_DATA_FORMAT_MASK            (0x7 << I2S_FMT_DATA_FORMAT_SHIFT)
+
+/** @brief Standard I2S Data Format.
+ *
+ * Serial data is transmitted in two’s complement with the MSB first. Both
+ * Word Select (WS) and Serial Data (SD) signals are sampled on the rising edge
+ * of the clock signal (SCK). The MSB is always sent one clock period after the
+ * WS changes. Left channel data are sent first indicated by WS = 0, followed
+ * by right channel data indicated by WS = 1.
+ *
+ * @verbatim
```

Comment:
Instead of using a @verbatim block, you can just indent the text drawing with 4 spaces. This formats the indented lines as a preformatted block and the leading `*` are stripped off.